src/eve/codegen.py

```diff
@@ -692,16 +690,14 @@ def generic_visit(self, node: TreeNode, **kwargs: Any) -> Union[str, Collection[
                         node=node,
                     ) from e.__cause__
 
-        elif isinstance(
-            node, (collections.abc.Sequence, collections.abc.Set)
-        ) and utils.is_collection(node):
-            result = [self.visit(value, **kwargs) for value in node]
-        elif isinstance(node, collections.abc.Mapping):
-            result = {key: self.visit(value, **kwargs) for key, value in node.items()}
-        else:
-            result = self.generic_dump(node, **kwargs)
+        elif isinstance(node, (list, tuple, set, collections.abc.Set)) or (
+            isinstance(node, collections.abc.Sequence) and not isinstance(node, (str, bytes))
```

Comment:
A good example of why I like nominal typing over duck typing

Reply:
At least it can be hidden in a method here ;)